### PR TITLE
Added emergency tracking option to suit sensors

### PR DESCRIPTION
**Pull request by MINIMAN10000**
Added emergency tracking option to suit sensors to alert users that you need attention before you die. People crew monitors prioritize those in distress and the only way to signify that with the current system is the die. This gives the player the option to flag up distress so that users of the crew monitor know to attempt a rescue.

In other words Emergency is tracking beacon mode but with orange "Alive" text and icon which signifies an emergency

[Video Example](https://puu.sh/zOnsB/1c0fe981b0.mp4)

clothing.dm adds the emergency option and associated text
crew_monitor.tmpl associates orange text with emergency option to crew monitor detail list
crew_monitor_map_content.tmpl associates orange text/icon to tracker map

:cl:
add: Added emergency tracking to suit sensors
/:cl:

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
Why do suit sensors and survivability need yet another buff at this stage? Things have quickly approached the point where it's increasingly difficult to die by non-perma kill methods and stay dead.

Crew Monitors, trackers, printable cameras, auto-cloning, the works.

Yes, a number of these I have supported, but there is a point where this does need to stop, otherwise it's going to eventually devolve into "someone took 32.532 damage in exactly 10 seconds; that means it's a changeling!" or other patently ridiculous scenarios.

So I'd like to ask what's the justification for this? Likewise, isn't the entire purposes defeated if EVERYONE sets their sensors to this instead of the level just below it? Why would why, as a player, ever want to choose "max sensors without emphasis to medical" over "max sensors with emphasis to medical"? This seems like a false choice in every situation.

**Comment by MINIMAN10000** [comment]
@FalseIncarnate  Yeah I reversed the URL/Text, went ahead and fixed that. 

As the video shows, when suit sensors are set to Emergency it has all the tracking features of tracking beacon "Alive" and tracker map icon orange. This being unique to "Emergency" suit sensor signifies distress.

**Comment by Fox-McCloud** [comment]
Hardly anyone is going to use them like that. They're just going to pick emergency since it stands out more, by default and is more likely to catch attention.  In "emergency situations" you don't usually have the time to set sensors yourself.

Fast forward a month or so, and everyone will be using it over regula maxed suit sensors.

**Comment by taukausanake** [comment]
I kind of like it but I agree with Fox here. If people dying is such a problem then maybe Medical should keep a better eye on the crew monitor and manifest. If people don't have sensors up, or if they aren't even showing up and not confirmed dead, then _just maybe_ something is up

**Comment by MINIMAN10000** [comment]
@Fox-McCloud Yes, like pulling a fire alarm for the purpose of causing a distraction setting Emergency suit sensors someone who deals with it may end up resulting in peoples time being wasted, or it may otherwise be a trap.

The justification for this goes back to why I set out to create this in first place. After entering virology after being infected with blood vomiting I end up cured of the blood vomiting the had gotten a new virus while in there, sensory disruption. I had alerted over comms that I needed salt to be cured by a virus but no one over comms acted. So I told the robot next to me he must get me salt, to which he assured me he got the guy next to me salt. I then collapsed slowly dying. I spent the next 10 minutes in virology having already alerted that individual as well as requested help of comms but to no avail. The robot then dragged me to the medicine cabinet leaving me vomiting for 10 minutes. Then I got moved into maintenance for radiation belt where he walked away and forgot my body where I laid for 20 minutes until I died. Once dead a person immediately showed up to retrieve my body... because people actually pay attention to suit sensors because they know people who are dead require attention, but people who are injured may just be making their way to medical.

If I could have flagged up an emergency after obtaining sensory disruption people who are rescuing players using portable crew monitor would know that I need to be moved to medical and get treatment.

An injured player isn't a sign that a player needs help unless it's something obvious like rapid suffocation damage.

With Emergency suit sensor the player is immediately aware that I won't be able to make my way to medical or otherwise require assistance.

All it does is change the color of icons and Alive tags to signifies that emergency help is needed it isn't superior to tracking beacon merely it signifies to your crew that you need their attention.

**Comment by Ty-Omaha** [comment]
I think it should flash

I also agree with how people will just set it on "Emergency" 24/7 by default

If anything this should be automatic

With that being said, if people are well enough to toggle their suit sensors they are well enough to call for help over the radio which is far more likely to get you help

**Comment by MINIMAN10000** [comment]
There are only 3 colors for icons. Red, Green, Orange. Flashing would most likely mean blinking between orange and green. I'm picturing the detail list being particularly annoying to anyone using a crew monitor, the point is to convey a message and the best way to do that is to be a unique solid color.

The point of emergency is that you are in a situation where you have the time to set your suit sensors but otherwise think you need getting to medical. Your bones are broken, if you flag up emergency you can have a paramedic show up and drag you to medical without you damaging your internals. You're bleeding and don't know if you can make it, so you flag up just in case you pass out. 

It allows the paramedic a quick way to know where he should be responding after he has dealt with all the dead bodies rather than watching a bunch of numbers and seeing if they move.

Many of the situations you want to prepare for the worst such as broken bones, bleeding out, a virus, are situations where automation just doesn't exist and creating such a setup you're back to Fox's concerns of "where does the automation stop"

Calling for help is worth a shot but if the paramedic has his hands full or the crew otherwise misses your message and you can't send another there's nothing more you can do. But your suit will stay on emergency mode until someone walks up you and turns it off.

**Comment by taukausanake** [comment]
You know what would make me support this no problem? No common radio headsets. I can only imagine in a real world setting how annoying comms must be and I don't really know why it exists on a space station where everyone has a job. The wall radios and station bound radios should fill that roll itself. The headsets would better be for specific departments so coordinating is easier. Without common radio I can see this being a lot more useful

**Comment by Aurorablade** [comment]
I am with fox on all his points here. Not really keen on this.

**Comment by Aurorablade** [comment]
"But your suit will stay on emergency mode until someone walks up you and turns it off." 
while antags turn off suit sensors, this really dosen't leave room for the antag to get away quick and have to move more lethally. Before the valid patrol with their crew monitors show up.

**Comment by MINIMAN10000** [comment]
Just take make it clear. Like shouting "help sci maint" you have to manually set suit sensors to Emergency.

**Comment by SilvernKell** [comment]
Yeah, but unlike shouting "help sci maint" it doesnt appear on many screens and you have to be actively looking for it. The time spent changing your sensors is time spent hoping someone was even looking and that the 50 other random dudes using it arent being helped first, or that you need sec or something instead of medical. 
It'd make more sense if swapping it to distress beacon made it call out on common but can you imagine how annoying it'd be? Maybe if you needed to be hurt as well for it to work. 
Either way it feels to me that "more automation" is not how you "fix" the scenario you described but "people being more competent" is how. The tools are already there it just depends on who is looking, how hard they are trying and how much they give a damn. 
The situation you described is annoying but if you try being a synthetic player practically every round feels like this tbh >_>;; It's not something worth being indignant over or trying to "fix" because people running around like useless headless chickens can be half the fun. Oh and when you're synthetic nobody bothers to immediately turn up to rez you either. You got it easy!

**Comment by Citinited** [comment]
Agree with all of what Fox said. We do not need to buff an already powerful system.

**Comment by Birdtalon** [comment]
The change is quite unnoticeable on the suit sensors window anyway. Plus everything that @Fox-McCloud said is true. Suit sensors are already strong.

No need to make the game even more hugboxy. If a player doesn't want to die then they have to make a conscious decision to stay in safer places. If they die regardless; that's the game.

File: code/modules/clothing/clothing.dm [hunk 1]
```diff
@@ -260,6 +262,9 @@ BLIND     // can't see anything
 			if(3)
 				for(var/mob/V in viewers(user, 1))
 					V.show_message("[user] sets [src.loc]'s sensors to maximum.", 1)
```
Comment (Citinited) [inline code/modules/clothing/clothing.dm]: Change maximum to tracking beacon here, as this isn't the maximum the sensors go to now.

File: code/modules/clothing/clothing.dm [hunk 2]
```diff
@@ -260,6 +262,9 @@ BLIND     // can't see anything
 			if(3)
 				for(var/mob/V in viewers(user, 1))
 					V.show_message("[user] sets [src.loc]'s sensors to maximum.", 1)
+			if(4)
+				for(var/mob/V in viewers(user, 1))
+					V.show_message("[user] sets [src.loc]'s sensors to emergency.", 1)
```
Comment (Citinited) [inline code/modules/clothing/clothing.dm]: `src` is implied here.

File: code/modules/clothing/clothing.dm [hunk 3]
```diff
@@ -224,7 +224,7 @@ BLIND     // can't see anything
 		to_chat(user, "This suit does not have any sensors.")
 		return 0
 
-	var/list/modes = list("Off", "Binary sensors", "Vitals tracker", "Tracking beacon")
+	var/list/modes = list("Off", "Binary sensors", "Vitals tracker", "Tracking beacon","Emergency")
```
Comment (Citinited) [inline code/modules/clothing/clothing.dm]: Needs a space between the comma and `"Emergency"`

**Comment by Tayyyyyyy** [comment]
This is a cool PR, but I don't think this will be useful. This is unnecessarily complicated. There's no need to add in new systems that will be only be used in a small fraction of situations. If there's an N2O gas leak in your room, you're probably going to spending the time getting out of there, instead of toggling your suit sensors to emergency mode and then passing out instead of getting away.

Also, people are going to forget to take themselves off. There's already like 20 suit sensor announcements in every round. People will just ignore this extra level.

If the problem is that it's hard to tell who's healthy and who needs medical attention at a glance, just add an orange "critical" for people who are in critical. This covers most of the cases where you can't get to medbay yourself but need help.

The UI/UX design of this feature just doesn't work.

**Comment by MINIMAN10000** [comment]
The crew monitor already reports vitals, you know who is critical because they have over 100 vital damage. Reporting that would be redundant and doesn't tell you anything new.

The idea is that the person who went to go save you can turn off or otherwise remind you to turn off your distress.

Like fire alarms, if an engineer gets an alert that there is a fire alarm pulled to check it out, maybe there is an emergency maybe there isn't if someone is causing trouble deal with the troublemaker. But we give people the option to declare the emergency because options is what space station 13 does best.

**Comment by Tayyyyyyy** [comment]
> The idea is that the person who went to go save you can turn off or otherwise remind you to turn off your distress.

There's so many opportunity for human error here...people forget to set their own sensors all the time. You're now expecting people to:

1. Not set their suit sensors to emergency all the time
2. Remember to set their suit sensors to emergency in the 1% of cases where it's needed, cause as you stated earlier, the vitals will already tell you that they need help in 99% of cases. There's going to be a very tiny sliver of cases where turning on emergency mode is better than getting yourself to medbay or getting to a central hallway and asking the nearest civy to take you there.
3. Remember to turn off other people's emergency mode, or for the person to turn it off themselves.

UI design that doesn't account for (widely common) human error is bound to fail

> But we give people the option to declare the emergency because options is what space station 13 does best.

Space station 13 is also super hard to learn because there are so many different, inconsistent, conflicting systems. "Keep it simple" is a great design principle to help reduce complexity creep. Introducing a highly visible "emergency mode" that is rarely useful only serves to confuse.

**Comment by SkeletalElite** [comment]
Just make it to where the suit sensors revert to normal automatically after one minute or so, because then no one would bother to redo it every minute and it still servers it's purpose.

**Comment by Kyep** [comment]
The issue with this is that it is exactly the same as full tracking mode, only better, and has no tradeoff.

If you want to add a mode that screams for attention even more than full sensors do, then it has to either be temporary (auto-resets to full after 60 seconds) or have some tradeoff.

If you simply want to add a new sensor mode, then it has to be more than just "full+".
For example, you might choose to add a mode that works like full when you're dead, but works like some lesser mode when you're alive. That way, for example, if you're an antagonist you can set your sensors to this mode so you have a chance of being revived if killed, but you can't be hunted down while you're alive using your sensors. That might be an interesting mode, but it still suffers from the potential issues of making it too easy to come back from death.

A better idea for a new sensor mode might be to go in the opposite direction. Make it like an existing mode, but more restrictive. For example, make it like full mode, but make the person only show up on full medical consoles, not on handheld crew monitors. Captain/heads might set their jumpsuits to this mode, so that they can be located in an emergency but they're harder to ambush. They trade some protection from common crew, in exchange for being harder to track by common crew.

Another idea for a new sensor mode might be "rigged" mode. The idea being that if you emag a jumpsuit you short out its life sensors and make it act like the person is always alive, even when they aren't. This could be used by antags to kill people without anyone noticing. Alternatively, to make it show them in an area they are not actually in. Essentially a less-functional / slightly sneaky sensor mode used by antags.

Heck, you could outright add an emagged sensor mode... emag a jumpsuit and its sensors no longer show up on regular crew monitors, but they DO show up on emagged crew monitors. That could allow antags (who make use of this) to track each other and work together more easily, whilst limiting the ability of crew to do the same. Hell, it could add interesting new gameplay like emagging someone (their jumpsuit, if exposed) to disable their sensors quickly but obviously.

In short: there are lots more creative things you could do with the concept of a new sensor mode - please don't make it just "full sensors, but better, with no downside".

**Comment by MINIMAN10000** [comment]
@Tayyyyyyy Yes like fire alarms people shouldn't have it on all the time, we don't run around the stations tripping all the fire alarms because we can unless we're trying to be a bother 

but I agree something should be done so people don't forget about it like any normal suit sensor setting. I believe a periodic message to the player with distress mode active would be the best way to make people not forget as well as not something you want all the time.

Also UI, it shouldn't be just another item in the drop down menu it'll complicate things of particular note being the bottom most setting in suit sensors is the intuitive answer when told "suit sensors to maximum please"

But that also means I need to learn more about coding in beyond because adding new UI elements is something I haven't done.

Example of separating UI elements

**Comment by Ty-Omaha** [comment]
the only reason fire alarms get turned off is because firedoors get in the way, and a red light gets in your face

**Comment by MINIMAN10000** [comment]
I feel the discussion here has brought a lot of useful ideas to the table that will require me to go back to work on my branch. Therefore this pull request no longer serves any purpose as it's current state is to simplified.
